src/index.js

```diff
@@ -157,7 +203,9 @@ var getKey = function(vdom) {
   return vdom == null ? null : vdom.key
 }
 
-var patch = function(parent, node, oldVNode, newVNode, listener, isSvg) {
+var patch = function(parent, node, oldVNode, newVNode, listener, actionMap, isSvg) {
+  actionMap = compose(actionMap, newVNode.actionMap)
```

Comment:
@zaceno This could affect performance. Can we avoid this?

Reply:
I don't think this will affect performance badly, because if you look at how `compose` is defined, there will only be a new function created when necessary (i e only once for each time `map` is used in the vdom).

We can't avoid it unfortunately, without taking the entire implemation of `map` out of core. I already demonstrated before that it is possible to implement the entire thing in userland. The problem with that is that it is quite inefficient since the vdom needs to be processed once to compose action maps, and then sent into `patch` for processing again.